mongo/integration/unified/collection_operation_execution.go

```diff
@@ -122,6 +122,10 @@ func executeBulkWrite(ctx context.Context, operation *operation) (*operationResu
 	}
 
 	res, err := coll.BulkWrite(ctx, models, opts)
+	if err != nil {
+		return newDocumentResult(nil, err), nil
+	}
```

Comment:
If `err` wasn't `nil` here, a non-nil `err` could've been overwritten on L133. Added logic to return the error (inside the result) immediately here.

Reply:
I don't know if we test it, but I believe bulkwrite can return both a  result and an error. We may want to store L133 err in a differently named variable rather than exiting early

Reply:
Good catch! Removed this check and renamed `err` on L133 to `marshalErr`. We'll rely on L148 to return the potential `err` from `BulkWrite` along with the `raw` result.